Moq.AutoMock.Tests/ResolvesSelf.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Moq.AutoMock.Tests
+{
+    [TestClass]
+    public class ResolvesSelf
+    {
+        [TestMethod]
+        public void Resolves_self_instance_directly()
+        {
+            AutoMocker mocker = new();
+
+            var resolved = mocker.Get<AutoMocker>();
+
+            Assert.IsTrue(ReferenceEquals(mocker, resolved));
+        }
+
+        [TestMethod]
+        public void Resolves_self_instances_when_a_dependency()
+        {
+            AutoMocker mocker = new();
+            mocker.With<HasAutoMockerDependency>();
+
+            var hasDependency = mocker.Get<HasAutoMockerDependency>();
+
+            Assert.IsTrue(ReferenceEquals(mocker, hasDependency.Mocker));
+        }
+
+        private class HasAutoMockerDependency
+        {
+            public HasAutoMockerDependency(AutoMocker mocker)
+            {
+                Mocker = mocker;
+            }
+
+            public AutoMocker Mocker { get; }
+        }
```

Comment:
Any reason this works or would be better?
```suggestion
        private record HasAutoMockerDependency(AutoMocker mocker);
```

Reply:
Nope, will fix.